tests/observability/scripts/perform_kiali_test.txt

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# Setup port-forwarding to Kiali pod as it is not accessible from outside
+# the cluster, but commands are run from outside the cluster.
+function setup_kiali_port_forwarding() {
+  local local_port=$1
+  local kiali_pod=$(kubectl get pod -l app=kiali -n istio-system | grep 'kiali' | head -1 | awk '{ print $1 }')
+  local kiali_port=$(kubectl get service kiali -n istio-system -o jsonpath='{.spec.ports[?(@.name=="http-kiali")].targetPort}')
+
+  echo "Setting up port-forwarding for Kiali at local port $local_port"
+  kubectl port-forward $kiali_pod $local_port:$kiali_port -n istio-system &
+  KIALI_PORT_FORWARD_PID=$!
+  KIALI_LOCAL_URL="http://localhost:$local_port/kiali/api"
+}
+
+function cleanup_kiali_port_forwarding() {
+  local exit_code=$?
+  echo "Exited with $exit_code"
+  echo "Stopping port-forward for Kiali"
+  if [ -z $KIALI_PORT_FORWARD_PID ]; then
+    echo 'It seems that Kiali port-forward PID is not set up properly'
+    kill $(netstat -ntpl | grep '127.0.0.1:8888' | awk '{ print $7 }' | cut -d'/' -f1)
+  else
+    kill $KIALI_PORT_FORWARD_PID
+  fi
+  exit $exit_code
+}
+
+## Setup port-forwarding for Kiali and register trap
+## for cleaning up on exit
+echo "Setting up port forwarding from 8888 to Kiali's API port"
+setup_kiali_port_forwarding 8888 # TODO: Pick a random unused port
+
+echo "Register clean up task for Kiali port-forward on exit"
+trap cleanup_kiali_port_forwarding EXIT
+
+# Allow port-forwarding to be setup in the background
+sleep 4
+
+
+## Login to Kiali using demo secret and save the token
+echo "Obtain token for accessing Kiali API"
+DEMO_SECRET='YWRtaW46YWRtaW4='
+API_TOKEN=$(curl -H "Authorization: Basic $DEMO_SECRET" $KIALI_LOCAL_URL/authenticate | jq -r '.token')
+if [ -z $API_TOKEN ]; then
+  echo 'Failed to obtain auth token for Kiali'
+  exit 1
+fi
+
+
+# Attaches JWT obtained from Kiali and send
+# API request to Kiali 
```

Comment:
I think that "Attaches" should be "Attach". Or if not, then I think that "send" should be "sends".